### PR TITLE
generate radon checkerboard

### LINKED ISSUE #20364: please add to gen_pattern.py the option to create radon checkerboard
##### System information (version)

- OpenCV => 4.5.2
- Operating System / Platform => Ubuntu 18.05
- Compiler => python

##### Detailed description

in your documentation there is an example of a fancy checkerboard with markers etc under findChessboardCornersSB() (see [here](https://docs.opencv.org/4.5.2/checkerboard_radon.png)). This pattern works particularily well in my experiments but gen_pattern can't create it. It would be really helpfull if you could add it.

### PR DESCRIPTION
**Pull request by AleksandrPanov**
Generate radon checkerboard for `findChessboardCornersSB` and update docs
resolves #20364

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Review by mshabunin — APPROVED** [review]

**Comment by alalek** [comment]
@AleksandrPanov Is this PR in draft status?

**Comment by AleksandrPanov** [comment]
Yes

**Comment by AleksandrPanov** [comment]
The current code works like this:
<details>
<summary>Current radon pattern view</summary>
 
 `-o out.svg -r 8 -c 9 -T radon_checkerboard -s 20.0 -u mm -m 0 0 0 1 2 0 4 4 4 5 8 7 8 0`

image
image

</details>
I'm also going to update the documentation.

**Comment by AleksandrPanov** [comment]
@mshabunin, @alalek pool request is ready for review

File: doc/pattern_tools/gen_pattern.py [hunk 1]
```diff
@@ -96,8 +166,10 @@ def main():
                         dest="page_width", type=float)
     parser.add_argument("-h", "--page_height", help="page height in units", default=argparse.SUPPRESS, action="store",
                         dest="page_height", type=float)
-    parser.add_argument("-a", "--page_size", help="page size, superseded if -h and -w are set", default="A4", action="store",
-                        dest="page_size", choices=["A0", "A1", "A2", "A3", "A4", "A5"])
+    parser.add_argument("-a", "--page_size", help="page size, superseded if -h and -w are set", default="A4",
+                        action="store", dest="page_size", choices=["A0", "A1", "A2", "A3", "A4", "A5"])
+    parser.add_argument("-m", "--markers", help="list of cells with markers for the radon checkerboard",
```
Comment (mshabunin) [inline doc/pattern_tools/gen_pattern.py:171]: I think marker format should be noted here too, something like 
>marker coordinates as list of numbers: `-m 1 2 3 4` means markers in cells [1, 2] and [3, 4]

File: doc/pattern_tools/gen_pattern.py [hunk 2]
```diff
@@ -70,6 +72,74 @@ def make_checkerboard_pattern(self):
...
+                if x % 2 == y % 2:
+                    corner_types, is_inside = self._get_type(x, y)
+                    if is_inside:
+                        square = SVG("rect", x=x * spacing + xspacing, y=y * spacing + yspacing, width=spacing,
+                                     height=spacing, fill="black", stroke="none")
+                    else:
+                        square = SVG("path", d=self._make_round_rect(x * spacing + xspacing, y * spacing + yspacing,
+                                                                     spacing, corner_types), fill="black", stroke="none")
+                    self.g.append(square)
+        if self.markers is not None:
```
Comment (mshabunin) [inline doc/pattern_tools/gen_pattern.py:129]: Perhaps marker array should be checked for correctness:
* size is even
* no duplicates (?)
* marker coordinates are in correct range (see example)

Example:
```.sh
python3 doc/pattern_tools/gen_pattern.py -T radon_checkerboard \
  -r 3 -c 3 -w 100 -h 100 \
  -m 1 1 2 2 3 3 -1 -1
```
out

File: doc/tutorials/calib3d/camera_calibration_pattern/camera_calibration_pattern.markdown [hunk 3]
```diff
@@ -36,6 +36,10 @@ create a circle board pattern in file acircleboard.svg with 7 rows, 5 columns an
 
         python gen_pattern.py -o acircleboard.svg --rows 7 --columns 5 --type acircles --square_size 10 --radius_rate 2
 
+create a radon checkerboard for [findChessboardCornersSB()](https://docs.opencv.org/master/d9/d0c/group__calib3d.html#gadc5bcb05cb21cf1e50963df26986d7c9) with markers in (7 4), (7 5), (8 5) cells:
```
Comment (mshabunin) [inline doc/tutorials/calib3d/camera_calibration_pattern/camera_calibration_pattern.markdown:39]: Doxygen should find function by its name, link is not necessary:
```suggestion
create a radon checkerboard for findChessboardCornersSB with markers in (7 4), (7 5), (8 5) cells:
```

**Review by mshabunin — APPROVED** [review]
LGTM + minor comments

File: doc/pattern_tools/gen_pattern.py (hunk 2 above)
Comment (AleksandrPanov) [reply doc/pattern_tools/gen_pattern.py:129]: added the check to main

File: doc/pattern_tools/gen_pattern.py (hunk 1 above)
Comment (AleksandrPanov) [reply doc/pattern_tools/gen_pattern.py]: added

File: doc/tutorials/calib3d/camera_calibration_pattern/camera_calibration_pattern.markdown (hunk 3 above)
Comment (AleksandrPanov) [reply doc/tutorials/calib3d/camera_calibration_pattern/camera_calibration_pattern.markdown]: also need to add parentheses `findChessboardCornersSB`, this done

**Review by mshabunin — APPROVED** [review]

**Pull request merged**